sdk/Pulumi/Deployment/Deployment_Run.cs

```diff
@@ -18,6 +18,15 @@ public partial class Deployment
         public static Task<int> RunAsync(Action action)
             => RunAsync(() =>
             {
+                var value = System.Environment.GetEnvironmentVariable("PULUMI_ATTACH_DEBUGGER");
+                if (value != null && value == "true")
+                {
+                    while (!System.Diagnostics.Debugger.IsAttached)
+                    {
+                        // keep waiting until the debugger is attached
+                        System.Threading.Tasks.Task.Delay(1);
```

Comment:
Need to await this

Reply:
We're not in an `async` context here, so I switched it out for `System.Threading.Thread.Sleep(1);`.  Or is there a strong reason why we should be using `Delay`?

Reply:
Delay is better if in an async context. Sleep works fine here.